controllers/kubevirtmachine_controller.go

```diff
@@ -349,6 +359,15 @@ func (r *KubevirtMachineReconciler) reconcileNormal(ctx *context.MachineContext)
 	return ctrl.Result{}, nil
 }
 
+func machineHasKnownInternalIP(kubevirtMachine *infrav1.KubevirtMachine) bool {
+	for _, addr := range kubevirtMachine.Status.Addresses {
+		if addr.Type == clusterv1.MachineInternalIP && addr.Address != "" {
```

Comment:
we should skip ipv6 link local

Reply:
That made sense to me at first, but after discussing this further I'm not 100% sure. It's theoretically possible someone might be using this in a network where link local addresses for VMs technically work.  It's just that the use case we're most familiar with (vms on pod network) wouldn't use link local.